src/inmanta/data/__init__.py

```diff
@@ -4728,6 +4807,95 @@ async def get_resource_deploy_summary(cls, environment: uuid.UUID) -> m.Resource
             results[row["status"]] = row["count"]
         return m.ResourceDeploySummary.create_from_db_result(results)
 
+    @classmethod
+    async def copy_resources_from_unchanged_resource_set(
+        cls,
+        environment: uuid.UUID,
+        source_version: int,
+        destination_version: int,
+        updated_resource_sets: abc.Set[str],
+        deleted_resource_sets: abc.Set[str],
+        *,
+        connection: Optional[asyncpg.connection.Connection] = None,
+    ) -> abc.Set[m.ResourceIdStr]:
+        """
+        Copy the resources that belong to an unchanged resource set of a partial compile,
+        from source_version to destination_version. This method doesn't copy shared resources.
+        """
+        query = f"""
+            INSERT INTO {cls.table_name()}(
+                environment,
+                model,
+                resource_id,
+                resource_type,
+                resource_id_value,
+                agent,
+                status,
+                attributes,
+                attribute_hash,
+                resource_set,
+                provides
+            )(
+                SELECT
+                    r.environment,
+                    $3,
+                    r.resource_id,
+                    r.resource_type,
+                    r.resource_id_value,
+                    r.agent,
+                    (
+                        CASE WHEN r.status='undefined'::resourcestate
+                        THEN 'undefined'::resourcestate
+                        ELSE 'available'::resourcestate
+                        END
+                    ) AS status,
+                    jsonb_set(r.attributes, '{{"version"}}'::text[], to_jsonb($3::integer)) AS attributes,
```

Comment:
this bit I don't understand?

Reply:
[Here](https://github.com/inmanta/inmanta-core/blob/971d2c6e29bb400b358a1a53d1138ac8734c093f/src/inmanta/server/services/orchestrationservice.py#L543) the attributes are filtered from the resource dictionary sent to the server. It only filters out the `id` field and not the version field, while I think it should. I guess this is a bug that already exists for quite a while. Do you agree this is a bug?

Reply:
I do, I would lift it out of the 'attributes' and use the one in the resource.